event/src/unvalidated/signed.rs

```diff
@@ -36,6 +36,26 @@ impl<D: serde::Serialize> Event<D> {
         )
     }
 
+    /// Factory for building an Event.
+    pub fn new(
```

Comment:
I started debating if we want to have a public new function like this (or if we could calculate the CIDs ourselves to avoid some potential errors), but we're going to have to turn it into a validated event at some point, which will involve checking all the CIDs and signatures, so it seems fine.